internal/kafka/internal/services/quota/quota_management_list_service.go

```diff
@@ -72,11 +94,19 @@ func (q QuotaManagementListService) ReserveQuota(kafka *dbapi.KafkaRequest, inst
 
 	var kafkas []*dbapi.KafkaRequest
 
+	billingModelID, err := q.detectBillingModel(kafka)
+	if err != nil {
+		return "", err
+	}
+	// TODO: find a better place to set this instead of this side effect
+	kafka.DesiredKafkaBillingModel = billingModelID
```

Comment:
Is it possible to move this at the start of this function or before this function is called? The error messages above will always return `0` max allowed instances if the billing model id is not specified in the request payload which can be a bit confusing when the quota configuration has the `max_allowed_instances` set to a non-zero number. 

Example error message received when testing this locally even though the `max_allowed_instances` of the 'standard' billing model was set to 2
```
"organization '13640203' has reached a maximum number of 0 allowed streaming units"
```

It will also avoid the scenario as described in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1368/files#r1024204938

Otherwise, we probably need to update the error message further down this function after the billing model has been detected.